query-translator/src/translate.rs

```diff
@@ -117,14 +123,23 @@ pub struct CombinedSelectQuery {
 }
 
 fn cc_to_select_query<T: Into<Option<u64>>>(projection: Projection, cc: ConjoiningClauses, limit: T) -> SelectQuery {
-    SelectQuery {
-        projection: projection,
-        from: FromClause::TableList(TableList(cc.from)),
-        constraints: cc.wheres
-                       .into_iter()
-                       .map(|c| c.to_constraint())
-                       .collect(),
-        limit: limit.into(),
+    if cc.is_known_empty {
+        SelectQuery {
```

Comment:
Ah, interesting -- further to my earlier comment, you do have the mechanism in place to statically represent a `NoneQuery`.  Consider lifting the semantic meaning of `FromClause::Nothing` up to the query level.  This would push your concrete `SelectQuery` to either be an enum, or to be a trait object in some form, and it might be that you don't want to do that for other reasons; I leave that to your discretion, since this will clearly work.